WordPress/wordcamp.org
# Speaker Feedback: Add UI for speakers to opt out of email notifications about approved feedback

Building on the usermeta value introduced in #441, this adds a toggle to the frontend Session Feedback view where speakers and organizers can disable and re-enable email notifications about newly approved feedback submissions. The toggle updates the usermeta value, which carries over across all WordCamp sites in the network.

To accomplish this, I created a new endpoint on our custom REST API route for feedback. The standard `users` endpoint doesn't work for updating this usermeta value because it requires that the user be a member on the current site. Most speakers are not added as members on the WordCamp sites that they speak at. The custom endpoint allows us to circumvent that.

Note that this does **not** include a metabox or other UI on the Speaker CPT screen for toggling this value. Adding a metabox that would be functional in the block editor turned out to be overly-complex, given the necessary interaction with the existing `WordPress.org Username` metabox field. Building a custom sidebar plugin is beyond the scope of this PR, but it might be simpler, and doable as a separate PR, once the other metabox fields for the Speaker CPT have been converted to sidebar panels, as was [recently done](https://github.com/WordPress/wordcamp.org/pull/303) for the Sessions CPT. In the mean time, organizers can toggle this value for speakers upon request by going to the frontend feedback view, where they will see the same UI that speakers do, but with a separate toggle for each speaker in the session.

Fixes #488 

### Screenshots

Speaker UI, notifications enabled (default)

![speaker-view-enabled](https://user-images.githubusercontent.com/916023/81439825-d912eb00-9123-11ea-935e-5255138ca204.jpg)

Speaker UI, notifications disabled

![speaker-view-disabled](https://user-images.githubusercontent.com/916023/81439837-e203bc80-9123-11ea-86ca-ba8c111ac7d6.jpg)

Organizer UI

![admin-view](https://user-images.githubusercontent.com/916023/81453076-7c242e80-913d-11ea-8351-338b46f337b3.jpg)

### How to test the changes in this Pull Request:

1. Rebuild the stylesheet to get the style updates.
1. With the same testing setup as for #441, open the feedback moderation screen and approve a new feedback submission.
1. Run the cron job and note that an email is sent in MailCatcher.
1. In a separate tab, navigate to the frontend feedback view for a session while logged in as a speaker. Click the toggle to disable notifications.
1. Approve another feedback submission and run the cron job. No new email should be sent.
1. Re-enable notifications for the speaker, and run the cron job again. Now the email should get sent.
